corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -413,6 +413,32 @@ hqDefine("geospatial/js/geospatial_map", [
             self.showFilterMenu = ko.observable(true);
             self.hasErrors = ko.observable(false);
 
+            // This function partitions user data into users with and without GPS data
+            function partitionUserData(userData) {
+                const userDataPartitions = {
+                    missingGPS: [],
+                    withGPS: {},
+                };
```

Comment:
Its nice that we got rid of iterating the same list twice though I guess we introduced more complexity in that process.
It's an extra overhead in reading/understanding this because we have one list and one object/collection here. It does not seem like its necessary for them to be different.

As a quick thought, I would think we would have simply two lists of cases, with /without GPS data or two collections with the properties "coordinates" and "link" can be filed or omitted if needed.
Additionally, it would be nice to see the logic being written only once for cases and users. This file is getting really long and could benefit from code duplication being reduced.